6.1: two `test_security_proxy` tests fails

## BUG/PROBLEM REPORT / FEATURE REQUEST

### What I did:

I ran tests for `zope.proxy` version 6.1.

### What I expect to happen:

All tests pass.

### What actually happened:

Two tests failed:
```
Failure in test test_security_proxy (zope.proxy.tests.test_proxy.Test_removeAllProxies)
Traceback (most recent call last):
  File "/usr/lib/python3.9/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.9/unittest/case.py", line 592, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
  File "/data/builds/ul-perl-modules/components/python/zope.proxy/build/prototype/i386/usr/lib/python3.9/vendor-packages/zope/proxy/tests/test_proxy.py", line 1337, in test_security_proxy
    self.assertIs(self._callFUT(proxy), c)
  File "/usr/lib/python3.9/unittest/case.py", line 1118, in assertIs
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python3.9/unittest/case.py", line 676, in fail
    raise self.failureException(msg)
AssertionError: <security proxied zope.proxy.tests.test_proxy.C instance at 0x7fffabab7670> is not <zope.proxy.tests.test_proxy.Test_py_removeAllProxies.test_security_proxy.<locals>.C object at 0x7fffabab7670>
```
```
Failure in test test_security_proxy (zope.proxy.tests.test_proxy.Test_sameProxiedObjects)
Traceback (most recent call last):
  File "/usr/lib/python3.9/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.9/unittest/case.py", line 592, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
  File "/data/builds/ul-perl-modules/components/python/zope.proxy/build/prototype/i386/usr/lib/python3.9/vendor-packages/zope/proxy/tests/test_proxy.py", line 1110, in test_security_proxy
    self.assertTrue(self._callFUT(proxy1, proxy1_2))
  File "/usr/lib/python3.9/unittest/case.py", line 688, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true
```

When I run tests for `zope.proxy` version 6.0 then all tests pass.

### What version of Python and Zope/Addons I am using:

Operating system: OpenIndiana
Python: 3.9.20
Zope.proxy: 6.1

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -5,6 +5,8 @@
 6.2 (unreleased)
 ================
 
+- Adjust minimal required version of ``zope.security`` test dependency to ``7.3``.
+
 
 6.1 (2024-10-02)
 ================

--- a/setup.py
+++ b/setup.py
@@ -135,7 +135,8 @@ setup(name='zope.proxy',
       zip_safe=False,
       extras_require={
           'test': [
-              'zope.security',  # We have a circular dependency for testing
+              # We have a circular dependency with zope.security for testing
+              'zope.security >= 7.3',
               'zope.testrunner',
           ],
           'docs': [

